doc/api/util.md

```diff
@@ -2515,14 +2515,46 @@ added: v10.0.0
 * `value` {any}
 * Returns: {boolean}
 
-Returns `true` if the value is an instance of a built-in [`Error`][] type.
+Returns `true` if the value was returned by the constructor of a
+[built-in `Error` type][].
 
 ```js
 util.types.isNativeError(new Error());  // Returns true
 util.types.isNativeError(new TypeError());  // Returns true
 util.types.isNativeError(new RangeError());  // Returns true
 ```
 
+Subclasses of the native error types will use the return value of the native
+error constructor as the new `this` and are therefore also native errors:
+
+```js
+class MyError extends Error {}
+util.types.isNativeError(new MyError());  // Returns true
+```
+
+A value being `instanceof` a native error is not equivalent to `isNativeError()`
+returning `true` for that value. Therefore, we recommend using
+`isNativeError(e) || e instanceof Error` to check if `e` is an error.
```

Comment:
I don't think that's good advice, typically to check if a value is an array, one should use `Array.isArray(value)`, I wouldn't recommend adding an `instanceof` check. In general, I think `instanceof` checks are rarely a good pattern in JS.

Reply:
First of all, thanks for the quick feedback! I applied all your suggestions.

Now to the recommended way of checking for errors:
I put this in there because this is what is done in [in Node.js internally](https://github.com/nodejs/node/blob/3c0131a4190a88211780dcc07dbaf84c8de97f34/lib/internal/util.js#L95).

Moreover, there are some popular libraries which produce errors that are not `isNativeError()`. I know of Axios (and have opened a PR there) and verror, but I'm pretty sure there are more. This is because people think that `Error.call(obj)` will turn `obj` into a native error, which it does not.

However, I am also not convinced that the `isNativeError(e) || e instanceof Error` check is the best idea because you could read [this section of the spec](https://tc39.es/ecma262/#sec-properties-of-error-instances) as a suggestion to use `Object.prototype.toString()` as the preferred way. This way of checkings also seems to have been the prefered by TC39 members [in a discussion 7 yers ago](https://github.com/ljharb/proposal-is-error). It also has the advantave of also working in the browser. However, this is exactly what the deprecated [util.isError()](https://nodejs.org/dist/latest-v19.x/docs/api/util.html#utiliserrorobject) function does.

Reply:
As the spec for [`%Object.prototype.toString%`](https://tc39.es/ecma262/#sec-object.prototype.tostring) states, its use to validate wether a value is one of the few specific kinds of built-in objects is considered legacy, and is not reliable because of `Symbol.toStringTag`.

The fact that `util.isError` is deprecated is another reason not to recommend doing this. (why would we keep the deprecation and in the same document recommend users to implement the deprecated method themselves?)

Reply:
So maybe just leave the recommendation out?